gnomad/utils/vcf.py

```diff
@@ -1167,24 +1243,32 @@ def make_hist_bin_edges_expr(
 
     :param ht: Table containing histogram variant annotations.
     :param hists: List of variant histogram annotations. Default is HISTS.
+    :param ann_with_hists: Name of row annotation containing histogram data. In exomes or
+        genomes release HT, `histograms` is a row, but in the joint releast HT, it's
+        under the row of `exomes`, `genomes`, or `joint`.
     :param prefix: Prefix text for age histogram bin edges.  Default is empty string.
     :param label_delimiter: String used as delimiter between prefix and histogram annotation.
     :param include_age_hists: Include age histogram annotations.
-    :return: Dictionary keyed by histogram annotation name, with corresponding reformatted bin edges for values.
+    :return: Dictionary keyed by histogram annotation name, with corresponding
+        reformatted bin edges for values.
     """
     # Add underscore to prefix if it isn't empty
     if prefix != "":
         prefix += label_delimiter
 
     edges_dict = {}
+    if ann_with_hists and hasattr(ht[ann_with_hists], "histograms"):
+        ht_row = ht.head(1)[ann_with_hists]
+    else:
+        ht_row = ht.head(1)
+
     if include_age_hists:
         edges_dict.update(
             {
                 f"{prefix}{call_type}": "|".join(
                     map(
                         lambda x: f"{x:.1f}",
-                        ht.head(1)
-                        .histograms.age_hists[f"age_hist_{call_type}"]
+                        ht_row.histograms.age_hists[f"age_hist_{call_type}"]
```

Comment:
This will fail if the datatype has missing values in the first row since you cannot index into a NoneType. We need to add logic to handle the missing values.

Reply:
head(1) worked very fast, my current change is very slow to run, do you have a faster suggestion?

Reply:
Unless I add more secondary workers...